packages/debugger/src/panels/sources/model.ts

```diff
@@ -16,13 +17,49 @@ export class SourcesModel implements IDebugger.Model.ISources {
    * @param options The Sources.Model instantiation options.
    */
   constructor(options: SourcesModel.IOptions) {
-    this.currentFrameChanged = options.currentFrameChanged;
+    this._currentSource = null;
+    this._currentFrame = null;
+
     this._displayRegistry =
       options.displayRegistry ?? new DebuggerDisplayRegistry();
+
+    if (options.mimeTypeService) {
+      this._mimeTypeService = options.mimeTypeService;
+    }
+
+    options.currentFrameChanged.connect(async (_, frame) => {
+      if (!frame) {
+        this._currentFrame = null;
+        this.currentSource = null;
+        return;
+      }
+
+      const displayPath = this.getDisplayName(frame);
+
+      const source = await options.getSource({
+        sourceReference: 0,
+        path: frame?.source?.path
+      });
+
+      const { content, mimeType } = source;
+      const editorMimeType =
+        mimeType ||
+        this._mimeTypeService.getMimeTypeByFilePath(frame.source?.path ?? '');
+
+      this._currentFrame = frame;
+
+      this.currentSource = {
+        content: content,
+        mimeType: editorMimeType,
+        path: displayPath
+      };
+    });
   }
 
   /**
    * Signal emitted when the current frame changes.
+   *
+   * @deprecated since 4.6.0, will be removed in 5.0.
```

Comment:
Maybe it could be worth adding this deprecation to the extension migration guide too: https://jupyterlab.readthedocs.io/en/latest/extension/extension_migration.html#jupyterlab-4-5-to-4-6-not-released-yet